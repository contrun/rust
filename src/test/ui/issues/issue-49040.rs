#![allow(unused_variables)]; //~ ERROR statements cannot reside in modules
//~^ ERROR `main` function
fn foo() {}
