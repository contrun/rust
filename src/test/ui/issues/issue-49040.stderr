error: statements cannot reside in modules
  --> $DIR/issue-49040.rs:1:28
   |
LL | #![allow(unused_variables)];
   |                            ^
   |
   = note: the program entry point starts in `fn main() { ... }`, defined in `main.rs`
   = note: for more on functions and how to structure your program, see https://doc.rust-lang.org/book/ch03-03-how-functions-work.html
help: consider moving the statements into a function
   |
LL | #![allow(unused_variables)]fn my_function() { }
   |                            ^^^^^^^^^^^^^^^^^^^^

error[E0601]: `main` function not found in crate `issue_49040`
  --> $DIR/issue-49040.rs:1:1
   |
LL | / #![allow(unused_variables)];
LL | |
LL | | fn foo() {}
   | |___________^ consider adding a `main` function to `$DIR/issue-49040.rs`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0601`.
